src/test/regress/sql/citus_local_tables_mx.sql

```diff
@@ -0,0 +1,148 @@
+SELECT create_citus_local_table('citus_local_table_1');
+CREATE TABLE reference_table_1(r1 int primary key);
+SELECT create_reference_table('reference_table_1');
+
+ALTER TABLE citus_local_table_1 ADD CONSTRAINT fkey_local_to_ref FOREIGN KEY(l1) REFERENCES reference_table_1(r1) ON DELETE CASCADE;
+
+CREATE TABLE citus_local_table_2(l1 int primary key);
+SELECT create_citus_local_table('citus_local_table_2');
+CREATE TABLE reference_table_2(r1 int);
+SELECT create_reference_table('reference_table_2');
+
+ALTER TABLE reference_table_2 ADD CONSTRAINT fkey_ref_to_local FOREIGN KEY(r1) REFERENCES citus_local_table_2(l1) ON DELETE RESTRICT;
+
+CREATE TABLE citus_local_table_3(l1 int);
+SELECT create_citus_local_table('citus_local_table_3');
+CREATE TABLE citus_local_table_4(l1 int primary key);
+SELECT create_citus_local_table('citus_local_table_4');
+
+ALTER TABLE citus_local_table_3 ADD CONSTRAINT fkey_local_to_local FOREIGN KEY(l1) REFERENCES citus_local_table_4(l1) ON UPDATE SET NULL;
+
+-- and switch to worker1
+\c - - - :worker_1_port
+SET search_path TO citus_local_tables_mx;
+
+-----------------------------------------------------------
+-- foreign key from citus local table to reference table --
+-----------------------------------------------------------
+
+-- show that on delete cascade works
+INSERT INTO reference_table_1 VALUES (11);
+INSERT INTO citus_local_table_1 VALUES (11);
+DELETE FROM reference_table_1 WHERE r1=11;
+-- should print 0 rows
+SELECT * FROM citus_local_table_1 ORDER BY l1;
+
+-- show that we are checking for foreign key constraint, below should fail
+INSERT INTO citus_local_table_1 VALUES (2);
+
+-- below should work
+INSERT INTO reference_table_1 VALUES (2);
+INSERT INTO citus_local_table_1 VALUES (2);
+
+-----------------------------------------------------------
+-- foreign key from reference table to citus local table --
+-----------------------------------------------------------
+
+-- show that we are checking for foreign key constraint, below should fail
+INSERT INTO reference_table_2 VALUES (4);
+
+-- below should work
+INSERT INTO citus_local_table_2 VALUES (4);
+INSERT INTO reference_table_2 VALUES (4);
+
+-------------------------------------------------------------
+-- foreign key from citus local table to citus local table --
+-------------------------------------------------------------
+
+-- show that we are checking for foreign key constraint, below should fail
+INSERT INTO citus_local_table_3 VALUES (3);
+
+-- below shoud work
+INSERT INTO citus_local_table_4 VALUES (3);
+INSERT INTO citus_local_table_3 VALUES (3);
+
+UPDATE citus_local_table_4 SET l1=6 WHERE l1=3;
+
+-- show that it prints only one row with l1=null due to ON UPDATE SET NULL
+SELECT * FROM citus_local_table_3;
+
+-- finally show that we do not allow defining foreign key in mx nodes
+ALTER TABLE citus_local_table_3 ADD CONSTRAINT fkey_local_to_local_2 FOREIGN KEY(l1) REFERENCES citus_local_table_4(l1);
```

Comment:
Lets add some savepoint regression tests at this moment. It shouldn't take a lot of time, you can check the existing savepoint tests.

To simple tests like, 

Do a modification to  citus local table;
rollback to savepoint;
see that the modification is rolled back

===

Do a modification to  citus local table;
Do a modification to  distributed table;
rollback to savepoint;
see that the modificatios are rolled back

===
Do a modification to  citus local table;
Do a modification to  reference table;
rollback to savepoint;
see that the modifications are rolled back